run_tests.py

```diff
@@ -0,0 +1,58 @@
+# This script is meant to carry out an automated test on all examples
+# of Nabla and log the console output of each to a text file which is
+# created and placed in the same directory as the executable of each example
+
+
+import glob, os, subprocess, time
+
+
+# walk into the examples_tests directory
+os.chdir("examples_tests")
+
+# list containing indices of each example to test
+# (I'm supposed to use list comprehension here but for some weird reason 'glob' refuse to work when I use it :\)
+indices = [1,2,3]
+
+# format each element of 'indices' to 2 digit
+def format_list():
+	for i in range(len(indices)):
+		indices[i] = str(indices[i])
+		indices[i] = '0' + indices[i]
+		if len(indices[i]) > 2:
+			indices[i] = indices[i].replace(indices[i][0], '', 1)
+
+
+def run_tests():
+	for i in range(len(indices)):
+		for fn in glob.glob(indices[i] + ".*"):
+		    print("\n", fn)
+		    os.chdir(fn)
+		    os.chdir("bin")
+
+		for fn in glob.glob("*.exe"):
+			p = subprocess.Popen(fn, stdout=subprocess.PIPE, shell=False, text=True)
+			try:
+				outs, errs = p.communicate(timeout=10)
+			except subprocess.TimeoutExpired:

```

Comment:
I know this is a temporary solution until we add "CI Mode" to our examples BUT not all of our examples have renderloop until the end of time. there are examples that test something and quit after a short time like `14.ComputeTest`.
That way the example quits but you don't log anything and don't consider it's return code.

Let me rephrase: `the timeout is only there to make sure we don't wait forever for the example to exit, but you must always assume that the example will quit after a short amount of time and consider it's return code`